pkg/kvstore/etcd.go

```diff
@@ -497,3 +497,44 @@ func (e *EtcdClient) Status() (string, error) {
 	}
 	return "Etcd: " + strings.Join(eps, "; "), err1
 }
+
+// StartWatch starts watching for changes in a prefix
+func (e *EtcdClient) StartWatch(w *Watcher) {
+	go func(w *Watcher) {
+		for {
+		recreateWatcher:
+			etcdWatch := e.cli.Watch(ctx.Background(), w.prefix, client.WithPrefix())
```

Comment:
Do you mean lastRevision before the reconnect? That may be potentially risky if we reconnect to an etcd cluster which had its state cleared. WithRev() is only really an optimization or do you see other value?